src/utils/media-utils.ts

```diff
@@ -0,0 +1,51 @@
+import axios from "axios"
+
+import { config } from "@config/config"
+
+import logger from "@root/logger/logger"
+import { MediaFileInput, MediaFileOutput } from "@root/types"
+import { getAccessToken } from "@root/utils/token-retrieval-utils"
+
+const GITHUB_ORG_NAME = config.get("github.orgName")
+
+export const isMediaFileOutput = (t: MediaFileOutput): t is MediaFileOutput =>
+  (t as MediaFileOutput).sha !== undefined
+
+export const getMediaFileInfo = async ({
+  file,
+  siteName,
+  directoryName,
+  mediaType,
+  isPrivate,
+}: MediaFileInput): Promise<MediaFileOutput> => {
+  const fileData = {
```

Comment:
there's some repeated code here, which makes it abit confusing to follow. wdyt about: 

1. shifting the encoding of `file.path` out into a util function (maybe `encodeFilePathAsUriComponent`)
2. either making another method that computes the github URL or making a util method that decides if sanitization is needed
3. shifting the shared properties out (only `mediaUrl` is diff b/w public/private) then using object spread rather than setting `.mediaUrl`

Reply:
Yeah refactoring makes sense, I've moved the main url generation into a separate method in https://github.com/isomerpages/isomercms-backend/pull/740/commits/7e317a77d5238c6f9f1ce57bcf4055f4cd531275